pilopt/src/lib.rs

```diff
@@ -123,6 +128,12 @@ impl ReferencedSymbols for EnumDeclaration {
     }
 }
 
+impl ReferencedSymbols for StructDeclaration {
```

Comment:
Can you also implement this for TypeDeclaration and then just call it there above?